### ISSUE TITLE
Automated taxes with UK nexus/shipping shows $0 when state/county is two words

### ISSUE THREAD

**Issue by yukikatayama** [Automattic/woocommerce-services#1852]
Ticket: 2550971-zen

**Example:**

Nexus: UK, NW10 1NU
Automated taxes enabled

When shipping to London, NW10 1NU or adding the county Dollis, tax calculates:

![](https://d.pr/i/O2SDe8+)
Link to screenshot: https://d.pr/i/O2SDe8

![](https://d.pr/i/AEkB16+)
Link to screenshot: https://d.pr/i/AEkB16

When making the county Dollis Hills (which is the correct name of the county), tax shows $0

![](https://d.pr/i/R5ErHN+)
Link to screenshot: https://d.pr/i/R5ErHN

Tested with US nexus/shipping address as well as Italy nexus/shipping, for example, but could not replicate. Seems to be specific to UK nexus/shipping.

Debug log for just Dollis:

<details>

```
12-10-2019 @ 12:49:31 - :::: TaxJar API called :::: (WCS Tax)
12-10-2019 @ 12:49:31 - Received: {"tax":{"taxable_amount":112.0,"tax_source":"destination","shipping":0.0,"rate":0.2,"order_total_amount":112.0,"jurisdictions":{"state":"DOLLIS","country":"GB","city":"LONDON"},"has_nexus":true,"freight_taxable":true,"breakdown":{"taxable_amount":112.0,"tax_collectable":22.4,"shipping":{"taxable_amount":0.0,"tax_collectable":0.0,"country_taxable_amount":0.0,"country_tax_rate":0.2,"country_tax_collectable":0.0,"combined_tax_rate":0.2},"line_items":[{"taxable_amount":112.0,"tax_collectable":22.4,"id":"32213-f5b0e808750108948ad43eadc5f6e5e9","country_taxable_amount":112.0,"country_tax_rate":0.2,"country_tax_collectable":22.4,"combined_tax_rate":0.2}],"country_taxable_amount":112.0,"country_tax_rate":0.2,"country_tax_collectable":22.4,"combined_tax_rate":0.2},"amount_to_collect":22.4}} (WCS Tax)
12-10-2019 @ 12:49:31 - :: Adding New Tax Rate :: (WCS Tax)
12-10-2019 @ 12:49:31 - {"tax_rate_country":"GB","tax_rate_state":"Dollis","tax_rate_name":"Dollis Tax","tax_rate_priority":1,"tax_rate_compound":false,"tax_rate_shipping":1,"tax_rate":20,"tax_rate_class":""} (WCS Tax)
12-10-2019 @ 12:49:31 - Tax Rate ID Set for 23 (WCS Tax)
12-10-2019 @ 12:49:31 - :: Tax Rate Found :: (WCS Tax)
12-10-2019 @ 12:49:31 - {"23":{"rate":20,"label":"Dollis Tax","shipping":"yes","compound":"no"}} (WCS Tax)
12-10-2019 @ 12:49:31 - :: Updating Tax Rate To :: (WCS Tax)
12-10-2019 @ 12:49:31 - {"tax_rate_country":"GB","tax_rate_state":"Dollis","tax_rate_name":"Dollis Tax","tax_rate_priority":1,"tax_rate_compound":false,"tax_rate_shipping":1,"tax_rate":20,"tax_rate_class":""} (WCS Tax)
12-10-2019 @ 12:49:31 - Tax Rate ID Set for 23 (WCS Tax)
```

</details>

Debug log for Dollis Hills:

<details>

```
12-10-2019 @ 12:49:37 - :::: TaxJar API called :::: (WCS Tax)
12-10-2019 @ 12:49:37 - Received: {"tax":{"taxable_amount":112.0,"tax_source":"destination","shipping":0.0,"rate":0.2,"order_total_amount":112.0,"jurisdictions":{"state":"DOLLIS HILL","country":"GB","city":"LONDON"},"has_nexus":true,"freight_taxable":true,"breakdown":{"taxable_amount":112.0,"tax_collectable":22.4,"shipping":{"taxable_amount":0.0,"tax_collectable":0.0,"country_taxable_amount":0.0,"country_tax_rate":0.2,"country_tax_collectable":0.0,"combined_tax_rate":0.2},"line_items":[{"taxable_amount":112.0,"tax_collectable":22.4,"id":"32213-f5b0e808750108948ad43eadc5f6e5e9","country_taxable_amount":112.0,"country_tax_rate":0.2,"country_tax_collectable":22.4,"combined_tax_rate":0.2}],"country_taxable_amount":112.0,"country_tax_rate":0.2,"country_tax_collectable":22.4,"combined_tax_rate":0.2},"amount_to_collect":22.4}} (WCS Tax)
12-10-2019 @ 12:49:37 - :: Adding New Tax Rate :: (WCS Tax)
12-10-2019 @ 12:49:37 - {"tax_rate_country":"GB","tax_rate_state":"Dollis Hill","tax_rate_name":"Dollis Hill Tax","tax_rate_priority":1,"tax_rate_compound":false,"tax_rate_shipping":1,"tax_rate":20,"tax_rate_class":""} (WCS Tax)
12-10-2019 @ 12:49:37 - Tax Rate ID Set for 24 (WCS Tax)
12-10-2019 @ 12:49:37 - :: Adding New Tax Rate :: (WCS Tax)
12-10-2019 @ 12:49:37 - {"tax_rate_country":"GB","tax_rate_state":"Dollis Hill","tax_rate_name":"Dollis Hill Tax","tax_rate_priority":1,"tax_rate_compound":false,"tax_rate_shipping":1,"tax_rate":20,"tax_rate_class":""} (WCS Tax)
12-10-2019 @ 12:49:37 - Tax Rate ID Set for 25 (WCS Tax)
12-10-2019 @ 12:49:37 - :::: TaxJar Plugin requested :::: (WCS Tax)
12-10-2019 @ 12:49:37 - :::: TaxJar API called :::: (WCS Tax)
12-10-2019 @ 12:49:37 - Received: {"tax":{"taxable_amount":112.0,"tax_source":"destination","shipping":0.0,"rate":0.2,"order_total_amount":112.0,"jurisdictions":{"state":"DOLLIS HILL","country":"GB","city":"LONDON"},"has_nexus":true,"freight_taxable":true,"breakdown":{"taxable_amount":112.0,"tax_collectable":22.4,"shipping":{"taxable_amount":0.0,"tax_collectable":0.0,"country_taxable_amount":0.0,"country_tax_rate":0.2,"country_tax_collectable":0.0,"combined_tax_rate":0.2},"line_items":[{"taxable_amount":112.0,"tax_collectable":22.4,"id":"32213-f5b0e808750108948ad43eadc5f6e5e9","country_taxable_amount":112.0,"country_tax_rate":0.2,"country_tax_collectable":22.4,"combined_tax_rate":0.2}],"country_taxable_amount":112.0,"country_tax_rate":0.2,"country_tax_collectable":22.4,"combined_tax_rate":0.2},"amount_to_collect":22.4}} (WCS Tax)
12-10-2019 @ 12:49:37 - :: Adding New Tax Rate :: (WCS Tax)
12-10-2019 @ 12:49:37 - {"tax_rate_country":"GB","tax_rate_state":"Dollis Hill","tax_rate_name":"Dollis Hill Tax","tax_rate_priority":1,"tax_rate_compound":false,"tax_rate_shipping":1,"tax_rate":20,"tax_rate_class":""} (WCS Tax)
12-10-2019 @ 12:49:37 - Tax Rate ID Set for 26 (WCS Tax)
12-10-2019 @ 12:49:37 - :: Adding New Tax Rate :: (WCS Tax)
12-10-2019 @ 12:49:37 - {"tax_rate_country":"GB","tax_rate_state":"Dollis Hill","tax_rate_name":"Dollis Hill Tax","tax_rate_priority":1,"tax_rate_compound":false,"tax_rate_shipping":1,"tax_rate":20,"tax_rate_class":""} (WCS Tax)
12-10-2019 @ 12:49:37 - Tax Rate ID Set for 27 (WCS Tax)
```

</details>

WCS: 1.22.1
WC: 3.8.1

**Comment by harriswong** [comment]
The space issue is related to [this](https://github.com/woocommerce/woocommerce/pull/25509#discussion_r372333496).

**Comment by harriswong** [comment]
Hi @yukikatayama, I believe [this is patched in 1.23.0](https://github.com/Automattic/woocommerce-services/issues/1600#issuecomment-586374077). The fix is [described here](https://github.com/woocommerce/woocommerce/pull/25509#issuecomment-581513178), and the change is [here](https://github.com/Automattic/woocommerce-services/pull/1899/files). 

I ran a quick test locally to verify; as well testing it against the new wc-admin UI.

1. I went to `http://localhost/wp-admin/admin.php?page=wc-settings&tab=tax&section=standard` to remove all existing "standard tax rate" that were created by Automated tax.
![image](https://user-images.githubusercontent.com/572862/107270760-952d7900-6a08-11eb-9724-fbf0757a1098.png)

2. I got 20% tax back
![image](https://user-images.githubusercontent.com/572862/107270827-a5455880-6a08-11eb-8a2b-afcbfb101b5b.png)

3. Back to "standard tax rate" `http://localhost/wp-admin/admin.php?page=wc-settings&tab=tax&section=standard`. I can see the tax being added automatically. 
![image](https://user-images.githubusercontent.com/572862/107270932-c4dc8100-6a08-11eb-8aed-70387c0b6e93.png)

I am closing this. Please let us know if this is still a problem and we can re-open it.
